src/move/emojicoin_arena/sources/emojicoin_arena.move

```diff
@@ -0,0 +1,713 @@
+            if (!registry_ref_mut.melee_ids_by_market_ids.contains(market_ids))
+                break;
+        };
+        market_ids
+
+    }
+
+    /// Pseudo-random proxy for `random_market_id` for use during `init_module`.
+    inline fun pseudo_random_market_id(n_markets: u64): u64 {
+        pseudo_randomness::u64_range(0, n_markets) + 1
+    }
+
+    /// Market IDs are 1-indexed.
+    inline fun random_market_id(n_markets: u64): u64 {
+        randomness::u64_range(0, n_markets) + 1
+    }
+
+    inline fun register_melee(
+        registry_ref_mut: &mut Registry,
+        n_melees_before_registration: u64,
+        sorted_unique_market_ids: vector<u64>
+    ) {
+        let melee_id = n_melees_before_registration + 1;
+        registry_ref_mut.melees_by_id.add(
+            melee_id,
+            Melee {
+                melee_id,
+                market_metadatas: sorted_unique_market_ids.map_ref(|market_id_ref| {
+                    option::destroy_some(
+                        emojicoin_dot_fun::market_metadata_by_market_id(*market_id_ref)
+                    )
+                }),
+                start_time: last_period_boundary(
+                    timestamp::now_microseconds(), registry_ref_mut.new_melee_duration
+                ),
+                duration: registry_ref_mut.new_melee_duration,
+                max_match_percentage: registry_ref_mut.new_melee_max_match_percentage,
+                max_match_amount: registry_ref_mut.new_melee_max_match_amount,
+                available_rewards: registry_ref_mut.new_melee_available_rewards,
+                all_entrants: smart_table::new(),
+                active_entrants: smart_table::new(),
+                exited_entrants: smart_table::new(),
+                locked_in_entrants: smart_table::new(),
+                n_melee_swaps: aggregator_v2::create_unbounded_aggregator(),
+                melee_swaps_volume: aggregator_v2::create_unbounded_aggregator(),
+                emojicoin_0_locked: aggregator_v2::create_unbounded_aggregator(),
+                emojicoin_1_locked: aggregator_v2::create_unbounded_aggregator()
+            }
+        );
+        registry_ref_mut.melee_ids_by_market_ids.add(sorted_unique_market_ids, melee_id);
+    }
+
+    inline fun remove_if_contains(
+        map_ref_mut: &mut SmartTable<u64, Nil>, key: u64
+    ) {
+        if (map_ref_mut.contains(key)) {
+            map_ref_mut.remove(key);
+        }
+    }
+
+    inline fun sort_unique_market_ids(market_id_0: u64, market_id_1: u64): vector<u64> {
+        if (market_id_0 < market_id_1) {
+            vector[market_id_0, market_id_1]
+        } else {
+            vector[market_id_1, market_id_0]
+        }
+    }
+
+    inline fun swap_with_stats<Emojicoin, LP>(
+        swapper: &signer,
+        swapper_address: address,
+        market_address: address,
+        input_amount: u64,
+        is_sell: bool,
+        integrator_fee_rate_bps: u8
+    ): (u64, u64) {
```

Comment:
It's not really a big deal and more of a readability thing here, but this is one of those cases where I think it could be clearer if you just made this into two functions, `sell_to_apt` and `buy_emojicoin`, since that's all the `is_sell` flag is ever used for.

Plus it (`isSell`) is passed as a raw `true`/`false` value, which is really non-descriptive

Reply:
Addressed in ce7856b6